### PR TITLE
add availability api.

### PR DESCRIPTION
**Pull request by sprixite**

### CODE REVIEW

File: app/views/mixins.py [hunk 1]
```diff
@@ -140,18 +140,38 @@ class FilterMixin(ListModelMixin):
     def get_queryset(self):
         emp_id_str = self.request.query_params.get("id")
         id = int(emp_id_str) if emp_id_str else None
+        date = self.request.query_params.get("date")
```
Comment (h-yaqub) [inline app/views/mixins.py:143]:
```suggestion
        interval = self.request.query_params.get("interval")
```

File: app/views/mixins.py [hunk 2]
```diff
@@ -140,18 +140,38 @@ class FilterMixin(ListModelMixin):
...
-        )
         end_date_str = self.request.query_params.get("end_date")
-        end_date = (
-            timezone.datetime.strptime(end_date_str, "%Y-%m-%d").date()
-            if end_date_str
-            else None
-        )
+
+        if not emp_id_str and not date and not start_date_str or not end_date_str:
+            print("ok")
```
Comment (h-yaqub) [inline app/views/mixins.py:148]:
```suggestion
```

File: app/views/mixins.py [hunk 3]
```diff
@@ -140,18 +140,38 @@ class FilterMixin(ListModelMixin):
...
+            return self.queryset.all()
+
+        if date == "weekly":
```
Comment (h-yaqub) [inline app/views/mixins.py:151]:
```suggestion
        if interval == "weekly":
```

File: app/views/mixins.py [hunk 4]
```diff
@@ -140,18 +140,38 @@ class FilterMixin(ListModelMixin):
...
+            end_date_str = timezone.now().date()
+            start_date_str = end_date_str - timedelta(days=7)
```
Comment (h-yaqub) [inline app/views/mixins.py:153]: We are asked to calculate the last week days. If above ran on Tuesday of current week, Monday of last week will be missed. Try below code:
```suggestion
            some_day_last_week = timezone.now().date() - timedelta(days=7)
            monday_of_last_week = some_day_last_week - timedelta(days=(some_day_last_week.isocalendar()[2] - 1))
            monday_of_this_week = monday_of_last_week + timedelta(days=7)
```

**Review by h-yaqub — CHANGES_REQUESTED** [review]

**Review by umairjameelprixite — APPROVED** [review]

**Pull request merged**